src/pages/KyberDAO/KNCUtility/GasRefundBox.tsx

```diff
@@ -0,0 +1,342 @@
+import { useTransactionAdder } from 'state/transactions/hooks'
+import { TRANSACTION_TYPE } from 'state/transactions/type'
+import { LinkStyledButton, MEDIA_WIDTHS } from 'theme'
+import { formattedNum } from 'utils'
+import { sendEVMTransaction } from 'utils/sendTransaction'
+
+import { readableTime } from '../Vote'
+import EligibleTxModal from './EligibleTxModal'
+import { KNCUtilityTabs } from './type'
+
+const Hr = styled.hr`
+  width: 100%;
+  border: none;
+  height: 1px;
+  background-color: ${({ theme }) => theme.border};
+  margin: 0;
+`
+
+const Wrapper = styled(Flex)`
+  width: 100%;
+  border-radius: 20px;
+  padding: 20px;
+  background-color: ${({ theme }) => theme.tableHeader};
+  gap: 28px;
+  flex-direction: column;
+`
+
+const Tab = styled(Text)<{ active?: boolean }>`
+  ${({ theme }) => theme.flexRowNoWrap}
+  align-items: left;
+  border-radius: 3rem;
+  outline: none;
+  cursor: pointer;
+  text-decoration: none;
+  color: ${({ theme }) => theme.subText};
+  font-size: 14px;
+  line-height: 20px;
+  font-weight: 500;
+
+  ${({ active, theme }) =>
+    active &&
+    css`
+      border-radius: 12px;
+      font-weight: 600;
+      color: ${theme.primary};
+    `}
+
+  :hover {
+    color: ${({ theme }) => darken(0.1, theme.primary)};
+  }
+`
+
+const Highlight = styled.span`
+  border-radius: 12px;
+  background: ${({ theme }) => transparentize(0.8, theme.primary)};
+  color: ${({ theme }) => theme.primary};
+  padding: 2px 8px;
+  display: inline-flex;
+  width: fit-content;
+  align-items: center;
+  gap: 8px;
+`
+
+export default function GasRefundBox() {
+  const { account, chainId } = useActiveWeb3React()
+  const { library } = useWeb3React()
+  const [selectedTab, setSelectedTab] = useState<KNCUtilityTabs>(KNCUtilityTabs.Available)
+  const theme = useTheme()
+  const { totalReward, reward, claimableReward } = useGasRefundInfo({ rewardStatus: selectedTab })
+  const toggleWalletModal = useWalletModalToggle()
+  const toggleEligibleTxModal = useEligibleTxToggle()
+  const openNetworkModal = useOpenNetworkModal()
+  const notify = useNotify()
+  const [claiming, setClaiming] = useState(false)
+  const addTransactionWithType = useTransactionAdder()
+  const eligibleTxs = useEligibleTransactions(1, 1)
+  const upToXXSmall = useMedia(`(max-width: ${MEDIA_WIDTHS.upToXXSmall}px)`)
+  const upToExtraSmall = useMedia(`(max-width: ${MEDIA_WIDTHS.upToExtraSmall}px)`)
+  const { userTier, gasRefundPerCentage } = useGasRefundTier()
+  const { daoInfo } = useVotingInfo()
+
+  const claimRewards = useCallback(async () => {
+    if (!account || !library || !claimableReward || claimableReward.knc <= 0) return
+
+    setClaiming(true)
+
+    const url = REWARD_SERVICE_API + '/rewards/claim'
+    const data = {
+      wallet: account,
+      chainId: chainId.toString(),
+      clientCode: 'gas-refund',
+      ref: '',
+    }
+    let response: any
+    try {
+      response = await axios({ method: 'POST', url, data })
```

Comment:
what do you think about rewardService.ts and using rtk here

Reply:
I clone this code from campaign :D If we create rtk `rewardService.ts`, might need some big refactor to apply them too.

Reply:
okay, noted for reward service refactor. tag @nhd98z for this